### PR TITLE
Change documentation theme to use `furo` theme

### PR DESCRIPTION
**Pull request by gunungpw**
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

- change default documentation from `python-docs-theme` to `furo`
- change `conf.py` to use `furo` default config first to look what need to change
- add `furo` and remove `python-docs-theme` dep to `doc/requirement.txt` 

Ref #5953

### CODE REVIEW

**Comment by gunungpw** [comment]
Preview can be seen in RTD workflow
## Dekstop
image

## Mobile
image

**Comment by DanielNoord** [comment]
I'm +1. But this is really something @Pierre-Sassoulas should decide.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think the "python doc theme" looks serious and "official" in the python world which is something that need to be taken into account. There's currently an uniformity between the python doc and pylint doc which is a good thing and trump the visual aspect imo. If python itself change its style I think we should follow.

**Comment by gunungpw** [comment]
Thank you, for your feedback.
From my point of view as a user and newly learn python language `python-docs-theme` does not have good usability compared to other themes like `material-mkdocs` or `furo`.

`furo` theme has a few good points:
- Navigation (next and previous button) is more approachable, I know what the next topic is
- Clean sidebar (Collapsible dropdown) not feel clutter.
- Content bar (right sidebar) gives a good indication about content on the page and navigating between topic/point
- Dark and light mode support
- Easy customizable (banner, logo, footer, edit button, font, accent color), [urrlib3 banner example](https://urllib3.readthedocs.io/en/stable/sponsors.html)
- use by many other popular libraries like;
    - [urllib3](https://urllib3.readthedocs.io/)
    - [attrs](https://www.attrs.org/)
    - [pip](https://pip.pypa.io/)
    - [Python Developer’s Guide](https://devguide.python.org/)
    - [psycopg3](https://www.psycopg.org/psycopg3/docs/)
    - [black](https://black.readthedocs.io/en/stable/)
- Have active maintainer, `furo` maintainer and creator was a member of pip and sphinx org

But, overall in my opinion `furo` just look cleaner and concise. Also official [Python Developer’s Guide](https://devguide.python.org/) hosted on python.org uses `furo`.

**Comment by cdce8p** [comment]
+1 from me. I've thought about doing this a while ago too. Never got around to doing it though. Usability wise `furo` is a really good option.

> I think the "python doc theme" looks serious and "official" in the python world which is something that need to be taken into account. There's currently an uniformity between the python doc and pylint doc which is a good thing and trump the visual aspect imo. If python itself change its style I think we should follow.

Not sure there is a need to stick with the official Python theme. Many "official" projects use other ones. `pip` is the best example. Consistency with astroid would be more important IMO.

File: doc/conf.py
Comment (Pierre-Sassoulas) [inline doc/conf.py:117]: I think we need to find a way to integrate what was in the doc previousely. Do we still have a link to github in the new style ?

File: doc/conf.py
Comment (Pierre-Sassoulas) [inline doc/conf.py:152]: What is the difference here ? It seems localtoc and globaltoc could be replaced by something generic from furo, but I'm not so sure for relatons and sourcelink.

Comment (gunungpw) [reply doc/conf.py:152]: `localtoc.html` will be replaced with content sidebar
image

`globaltoc.html` will be replaced with generic `furo` left sidebar:
image

Comment (gunungpw) [reply doc/conf.py:152]: `relation.html` will be replaced with generic `furo` navigation next and previous button:
- furo
image
- python-docs-theme

Comment (gunungpw) [reply doc/conf.py:152]: `sourcelink.html` will be replaced with edit on GitHub button (little pen icon), and because `pylint` uses `readthedocs`. Edit source file can also be seen in RTD edit/download/version options
image

File: doc/conf.py
Comment (gunungpw) [reply doc/conf.py:117]: `"collapsiblesidebar": True` -> Furo by default do this
`"issues_url": "https://github.com/pycqa/pylint/issues/new"` -> By default furo will show Github and RTD link in `footer`
image


```
"root_name": "PyCQA",
"root_url": "https://meta.pycqa.org/en/latest/",
```
these 2 options i am not familiar with, but `"https://meta.pycqa.org/en/latest/"` go to not exist page

File: doc/conf.py
Comment (gunungpw) [reply doc/conf.py:152]: [Furo sidebar docs](https://pradyunsg.me/furo/customisation/sidebar/)

File: doc/conf.py
Comment (gunungpw) [reply doc/conf.py:117]: [furo edit button](https://pradyunsg.me/furo/customisation/edit-button/)
[furo footer](https://pradyunsg.me/furo/customisation/footer/)

**Review by Pierre-Sassoulas — APPROVED** [review]
Thanks for the detailled explanation @gunungpw ! LGTM

**Comment by gunungpw** [comment]
Thank you, I learning so much here :)

File: doc/conf.py
Comment (DanielNoord) [inline doc/conf.py:111]: @gunungpw Would you like to do a follow-up to remove these options? 😄

Comment (gunungpw) [reply doc/conf.py:111]: Okay 👍

**Pull request merged**
